src/plugins/irc/irc-channel.c

```diff
@@ -549,22 +549,21 @@ irc_channel_remove_away (struct t_irc_server *server,
  */
 
 void
-irc_channel_check_away (struct t_irc_server *server,
+irc_channel_check_whox (struct t_irc_server *server,
                         struct t_irc_channel *channel)
 {
     if ((channel->type == IRC_CHANNEL_TYPE_CHANNEL) && channel->nicks)
     {
         if (server->cap_away_notify
+            || server->cap_account_notify
             || ((IRC_SERVER_OPTION_INTEGER(server, IRC_SERVER_OPTION_AWAY_CHECK) > 0)
                 && ((IRC_SERVER_OPTION_INTEGER(server, IRC_SERVER_OPTION_AWAY_CHECK_MAX_NICKS) == 0)
                     || (channel->nicks_count <= IRC_SERVER_OPTION_INTEGER(server, IRC_SERVER_OPTION_AWAY_CHECK_MAX_NICKS)))))
         {
-            channel->checking_away++;
+            channel->checking_whox++;
             irc_server_sendf (server, IRC_SERVER_SEND_OUTQ_PRIO_LOW, NULL,
-                              "WHO %s", channel->name);
+                              "WHO %s %%cuhsnfdar", channel->name);
         }
-        else
-            irc_channel_remove_away (server, channel);
```

Comment:
Why did you remove the call to irc_channel_remove_away() ?
I think if you remove this line, the nicks will never be reset to normal color (in nicklist) if you turn off away checking option.